### PR TITLE
Add connectionFactory getter

### LINKED ISSUE #358: Add RabbitMessageChannelBinder#connectionFactory getter
The `RabbitMessageChannelBinder` has an opinionated view about the `connectionFactory` lifecycle. 
For some use-cases a user might want to reset the connection; let's say the primary broker failed and we switched automatically to a secondary broker; later, when the primary broker is available, the user might want to (manually) switch back by resetting the connection to the secondary broker. (Kudos to @garyrussell)

### PR DESCRIPTION
**Pull request by tzolov**
Resolves #358

### CODE REVIEW

File: spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java [hunk 1]
```diff
@@ -277,6 +277,10 @@ public void destroy() throws Exception {
 		}
 	}
 
+	public ConnectionFactory getConnectionFactory() {
```
Comment (garyrussell) [inline spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java:280]: Javadocs?

**Review by garyrussell — CHANGES_REQUESTED** [review]
I suggest we backport this to 3.2.x as well.

File: spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java (hunk 1 above)
Comment (tzolov) [reply spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java:285]: addressed

**Review by garyrussell — APPROVED** [review]

**Comment by sobychacko** [comment]
@tzolov Any chance, you can issue this PR against the core repository? https://github.com/spring-cloud/spring-cloud-stream. Rabbit binder is now part of the mono-repo structure (for `main` and `3.2.x`).

**Comment by tzolov** [comment]
@sobychacko here it is https://github.com/spring-cloud/spring-cloud-stream/pull/2315

**Pull request closed without merging**